File: leetcode/problems/cpp/running-sum-of-1d-array.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(1)

#include "vector"
using namespace std;


class Solution {
public:
  vector<int> runningSum(vector<int>& nums) {
    int len = nums.size();
    for (int i = 1; i < len;i++)
      nums[i] += nums[i - 1];
    return nums;
  }
};

File: leetcode/problems/rs/running-sum-of-1d-array.rs
// by @codeAbinash
// Time : O(n)
// Space : O(n)

impl Solution {
    pub fn running_sum(nums: Vec<i32>) -> Vec<i32> {
        let mut result = Vec::with_capacity(nums.len());
        let mut sum = 0;
        for num in nums {
            sum += num;
            result.push(sum);
        }
        result
    }
}
